### ISSUE TITLE
ActionController::RoutingError (uninitialized constant ImpersonationController

### ISSUE THREAD

**Issue by novalisek** [rgtk/redmine_impersonate#4]
Hello,
I am using Redmine 3.4.3.devel.17016, Ruby 2.3.5, Rails 5.1.2 and when I try to impersonate, I always get this error in RM log:
----
```
Started POST "/admin/impersonation?user_id=3" for xxx.xxx.xxx.xxx at 2017-10-30 14:17:45 +0100

ActionController::RoutingError (uninitialized constant ImpersonationController
Did you mean?  ImportsController):

activesupport (5.1.2) lib/active_support/inflector/methods.rb:269:in `const_get'
activesupport (5.1.2) lib/active_support/inflector/methods.rb:269:in `block in constantize'
activesupport (5.1.2) lib/active_support/inflector/methods.rb:267:in `each'
activesupport (5.1.2) lib/active_support/inflector/methods.rb:267:in `inject'
activesupport (5.1.2) lib/active_support/inflector/methods.rb:267:in `constantize'
activesupport (5.1.2) lib/active_support/dependencies.rb:582:in `get'
activesupport (5.1.2) lib/active_support/dependencies.rb:613:in `constantize'
actionpack (5.1.2) lib/action_dispatch/http/request.rb:82:in `controller_class'
actionpack (5.1.2) lib/action_dispatch/routing/route_set.rb:43:in `controller'
actionpack (5.1.2) lib/action_dispatch/routing/route_set.rb:29:in `serve'
actionpack (5.1.2) lib/action_dispatch/journey/router.rb:46:in `block in serve'
actionpack (5.1.2) lib/action_dispatch/journey/router.rb:33:in `each'
actionpack (5.1.2) lib/action_dispatch/journey/router.rb:33:in `serve'
actionpack (5.1.2) lib/action_dispatch/routing/route_set.rb:832:in `call'
rack-openid (1.4.2) lib/rack/openid.rb:98:in `call'
request_store (1.0.5) lib/request_store/middleware.rb:9:in `call'
rack (2.0.3) lib/rack/etag.rb:25:in `call'
rack (2.0.3) lib/rack/conditional_get.rb:38:in `call'
rack (2.0.3) lib/rack/head.rb:12:in `call'
rack (2.0.3) lib/rack/session/abstract/id.rb:232:in `context'
rack (2.0.3) lib/rack/session/abstract/id.rb:226:in `call'
actionpack (5.1.2) lib/action_dispatch/middleware/cookies.rb:613:in `call'
actionpack (5.1.2) lib/action_dispatch/middleware/callbacks.rb:26:in `block in call'
activesupport (5.1.2) lib/active_support/callbacks.rb:97:in `run_callbacks'
actionpack (5.1.2) lib/action_dispatch/middleware/callbacks.rb:24:in `call'
actionpack (5.1.2) lib/action_dispatch/middleware/debug_exceptions.rb:59:in `call'
actionpack (5.1.2) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.2) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.2) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.2) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.2) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.2) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.2) lib/rails/rack/logger.rb:24:in `call'
actionpack (5.1.2) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.2) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/method_override.rb:22:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.2) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.2) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.2) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/content_length.rb:15:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.2) lib/rails/engine.rb:522:in `call'
rack (2.0.3) lib/rack/handler/webrick.rb:86:in `service'
/usr/lib/ruby/2.3.0/webrick/httpserver.rb:140:in `service'
/usr/lib/ruby/2.3.0/webrick/httpserver.rb:96:in `run'
/usr/lib/ruby/2.3.0/webrick/server.rb:296:in `block in start_thread'
```
Thanks for any tips.

**Comment by nounder** [comment]
It looks like part of plugin code is not where it supposed to be. What’s the content of $REDMINE_ROOT/plugins/redmine_impersonate/app/controllers directory?

You can try removing plugin and moving it back again. Maybe something went wrong durning moving files.

**Comment by novalisek** [comment]
Hello,
this is directory tree of plugins/redmine_impersonate:
````
./app
./app/controllers
./app/controllers/impersonation_controller.rb
./lib
./lib/redmine_impersonate.rb
./lib/redmine_impersonate
./lib/redmine_impersonate/hook.rb
./init.rb
./config
./config/locales
./config/locales/en.yml
./config/routes.rb
./NEWS
./LICENSE
./README.md
./test
./test/test_helper.rb
./test/integration
./test/integration/impersonation_test.rb
````
I already tried to delete the directory and git clone again

**Comment by novalisek** [comment]
Hello, I updated ruby to v2.4.2, checked out new Redmine from SVN rev17022 and did completely new installation, gem update, bundle update, installed just this plugin, but problem still remains :(

**Comment by nounder** [comment]
I've just downloaded this 2.4.3, configured it, installed plugin and still cannot reproduce reported error -- everything seems to work fine. 
At least I found and fixed little bug (3920cdbb27cce28f99c62b531a99948460c6e013) :dancer: 

Your Redmine doesn't seem to recognize this plugin code structure. I am pretty sure this is caused by environment misconfiguration. Maybe you can ask someone with Rails experience to look at it. Or you can just use other plugin with similar functionality as redmine_impersonate.

**Comment by nounder** [comment]
No reply. Closing.
